common/src/main/java/com/wynntils/services/itemfilter/type/StatFilterFactory.java

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under LGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.services.itemfilter.type;
+
+import java.util.Optional;
+
+/**
+ * A factory for creating {@link StatFilter} instances.
+ * @param <T>
+ */
+public interface StatFilterFactory<T> {
```

Comment:
This PR reverted the long discussion we had, where the initial "ItemFilterFactory + ItemFilter" became "ItemFilter + ItemFilterInstance".

I think this PR makes the same mistake all over. The problem is that we get a discrepancy between the concepts of the code and the concepts of the user. You would not describe "lvl" as a "filter factory" to the user, instead you describe it as a "filter". The user realizes they can put in several filters, so the idea of multiple "filter instances" is inherent in the user conceptual view (even if they perhaps would not use that exact terminology).

By calling the filter instances for "filter", and the abstract concept of the filter for "filter factory", we get the word "filter" to mean to very different things. This is just bad, and a source of confusion.

Was there any particular reason why you did not simply just change "Item" to "Stat"? (I'm not sure I think even that rename was helpful; what else about an item could you filter than it's "stats"?) Or did you just disagree with the previous resolution of renaming "filter + filter factory" to "filter instance + filter"?

Reply:
Perhaps move to discord? Seems like a better place to discuss

Reply:
To be clear: I fully understand and support the idea of separating the parsing of filters with their specific types (numerical and string) with the value from the item ("stat") to compare them to

The thing I am not so happy about is how this is expressed in names and concept. 

For instance, "StatProviderAndFilterPair" is quite a mouthful, and still does not describe what it does, only how it is constructed. This is basically what was previously called "ItemFilterInstance", which I believe is a much better name. I think this can be refactored back to much of the original names and concepts, while still keeping how the filter is split between parser and value providers.